stable/database/files/nuobackup

```diff
@@ -175,82 +172,28 @@ if [ -n "$semaphore" ]; then
 
 
    if [ -z "$gate" ]; then
-      echo >&2 "Semaphore $semaphore/$db_name empty - nothing to do"
+      echo "Semaphore $semaphore/$db_name empty - nothing to do"
       exit 0
    fi
 
    # wait for the database to be RUNNING
    nuocmd check database --db-name $db_name --check-running --wait-forever
 fi
 
-# find the index for the latest backup info
-latest=$(nuocmd get value --key $NUODB_BACKUP_KEY/$db_name/${backup_group}/latest )
-current=$latest
-
-if [ "$backup_type" = "full" ]; then
-   if [ -z "$latest" ]; then
-      current=0
-   fi
-
-   # bump latest, and wrap around (simple ring-buffer)
-   next=$(( (current + 1) % $NUODB_MAX_BACKUP_HISTORY ))
-else
-   if [ -z "$latest" ]; then
-      echo >&2 "No full backup has completed yet - try again later"
-      exit -1
+if [ "$backup_type" != "full" ]; then
+   error="$( nuodocker get current-backup \
+      --db-name "${db_name}" \
+      --labels "backup ${backup_group} ${labels}" 2>&1)"
+   if [ $? -ne 0 ] && echo "$error" | grep -q "No existing backup path found"; then
+      # Check if full backup has been started already for all archiveIds
+      # This will most likely execute in the following scenarious:
+      # * initial database backup request
+      # * backup request after the database have been restored
+      # * new HC SM has been started since the last full backup
+      echo "Executing full hotcopy as a prerequisite for ${backup_type} hotcopy: ${error}"
+      nuocmd check database --db-name $db_name --check-running --wait-forever
```

Comment:
This command will hang forever if an SM has been shut down, since the database will be in AWAITING_ARCHIVE_HISTORY_INC state, and _not_ the RUNNING state.

Reply:
Good point! You are correct that `check database` won't succeed if the database state is other than `RUNNING`.

>if any SM has been shut down
KAA will remove the archive for scaled-down replicas, hence, the database will be in `RUNNING` state even if SM has been shutdown.
Even if KAA is disabled, we won't get into the _if block_ as `get current_backup` is checking archives for registered domain processes only.

>or is failing to start
We might get into a situation where the `check database` is stuck if:
* Database is running
* HC SM statefulset is scaled up but the SM fails to start
* The failed SM is left into a crash-looping state.

The only purpose of the check here is to prevent several full backups after initial deployment.
We could just add a timeout of 300sec so that this doesn't block forever.